app/assets/javascripts/details/shell/c2.js

```diff
@@ -263,6 +264,20 @@ C2 = (function() {
     });
   }
 
+  C2.prototype._setupFormSubmitModal = function(){
+    var self = this,
+      events = "attachment_confirm-modal:confirm observer_confirm-modal:confirm";
+    this.modals.el.on(events, function(event, item, sourceEl){
+      self._submitAndClose(sourceEl);
+    });
+  }
+
+  C2.prototype._submitAndClose = function(sourceEl){
+    var self = this;
+    $(sourceEl).parent().submit();
```

Comment:
@nickbristow In the future, it might be worth changing this into a `$form.serialized()`.